src/framework/net/protocol.cpp

```diff
@@ -144,29 +144,35 @@ void Protocol::internalRecvData(uint8* buffer, uint16 size)
 
 void Protocol::generateXteaKey()
 {
-    std::mt19937 eng(std::time(nullptr));
-    std::uniform_int_distribution<uint32> unif(0, 0xFFFFFFFF);
-    m_xteaKey[0] = unif(eng);
-    m_xteaKey[1] = unif(eng);
-    m_xteaKey[2] = unif(eng);
-    m_xteaKey[3] = unif(eng);
+    std::random_device rd;
+    std::uniform_int_distribution<uint32> unif;
+    std::generate(m_xteaKey.begin(), m_xteaKey.end(), [&]() { return unif(rd); });
```

Comment:
I changed this too because the current algorithm is shit. Seeding with timestamp, passing default values as params anyway, and an unnecessary `mt19937` there when the server itself generates plenty of entropy for the `random_device`.